Project1/logic_pair.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+template<typename T1, typename T2>
+class logic_pair :
+	public std::pair<T1, T2> {
+public:
+	using std::pair<T1, T2>::pair;
+	using std::pair<T1, T2>::operator=;
+	using std::pair<T1, T2>::swap;
+	using std::pair<T1, T2>::first;
+	using std::pair<T1, T2>::second;
+	
+	template<typename pair>
+	bool operator==(const pair& other) const {
+		return first == other.first && second == other.second;
+	}
+	template<typename pair>
+	bool operator!=(const pair& other) const {
+		return first != other.first || second != other.second;
+	}
+	template<typename pair>
+	bool operator<(const pair& other) const {
```

Comment:
Nitpick: These don't short-circuit and it would be nice if they did

Also, you should be using logical-or `||`, not bitwise-or `|`

Reply:
short circuit?

Reply:
Doing that hurts my brain, but I think I 'fixed' it